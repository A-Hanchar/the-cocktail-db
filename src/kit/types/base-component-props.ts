export type BaseComponentProps<P = unknown> = P & {testID?: string};
