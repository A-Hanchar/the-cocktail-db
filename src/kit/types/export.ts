export * from './base-component-props';
export * from './tag-name';
export * from './generic-html-element';
