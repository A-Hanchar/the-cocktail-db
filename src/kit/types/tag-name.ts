import {JSX} from 'react';

export type TagName = keyof JSX.IntrinsicElements;
