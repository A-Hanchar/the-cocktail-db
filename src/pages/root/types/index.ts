import {CocktailCode} from '@cocktailDB/api';

export type CocktailPageModel = {
	code: CocktailCode;
	label: string;
};
