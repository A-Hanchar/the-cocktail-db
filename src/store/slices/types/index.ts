export enum ASYNC_STATUS {
	IDLE = 'IDLE',
	PENDING = 'PENDING',
	SUCCEEDED = 'SUCCEEDED',
	FAILED = 'FAILED',
}
