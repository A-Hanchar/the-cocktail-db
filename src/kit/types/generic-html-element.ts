import {DetailedHTMLProps, HTMLAttributes, JSX, ReactHTMLElement} from 'react';

import {TagName} from './tag-name';

export type GenericHTMLElement<T extends TagName> =
	JSX.IntrinsicElements[T] extends DetailedHTMLProps<HTMLAttributes<infer E>, infer E>
		? E
		: ReactHTMLElement<HTMLElement>;
