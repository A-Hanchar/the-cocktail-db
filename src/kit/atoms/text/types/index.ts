import {JSX, PropsWithChildren} from 'react';

import {BaseComponentProps} from '../../../types/base-component-props';
import {GenericHTMLElement} from '../../../types/generic-html-element';

export type TextTag = keyof Pick<JSX.IntrinsicElements, 'p' | 'span' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'>;

export type TextSize = '36px' | '24px' | '16px';

export type TextProps<T extends TextTag = TextTag> = BaseComponentProps &
	PropsWithChildren<{
		as: T;
		className?: string;
		weight?: '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900';
		uppercase?: boolean;
		size?: TextSize;
		color?: 'base' | 'light' | 'negative' | 'yellow' | 'inherit' | 'currentColor';
		ref?: React.Ref<GenericHTMLElement<T>>;
	}>;
