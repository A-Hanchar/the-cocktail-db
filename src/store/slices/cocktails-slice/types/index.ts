import {CocktailAlcoholic, CocktailCategory, CocktailCode} from '@cocktailDB/api';

import {ASYNC_STATUS} from '../../types';

export type CocktailIngredientMeasure = {
	ingredient: string;
	measure: string | null;
};

export type CocktailModel = {
	id: string;
	strDrink: string;
	strAlcoholic: CocktailAlcoholic;
	strCategory: CocktailCategory;
	strGlass: string;
	ingredientsWithMeasure: CocktailIngredientMeasure[];
	strDrinkThumb: string;
	strInstructions: string | null;
};

export type CocktailsStateModel = {
	status: ASYNC_STATUS;
	data: CocktailModel[];
};

export type CocktailsState = Record<CocktailCode, CocktailsStateModel>;
