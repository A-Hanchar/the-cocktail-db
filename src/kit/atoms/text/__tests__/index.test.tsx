import {render, screen} from '@testing-library/react';

import {Text} from '../';
import {TextTag} from '../types';

describe('<Text />', () => {
	it('SHOULD match snapshot', () => {
		const {asFragment} = render(<Text as='h1' />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('SHOULD render component with default testID', () => {
		const {getByTestId} = render(<Text as='p' />);

		expect(getByTestId('text')).toBeInTheDocument();
	});

	it('SHOULD render component with passed testID', () => {
		const testID = 'testID';

		const {getByTestId} = render(<Text as='p' testID={testID} />);

		expect(getByTestId(testID)).toBeInTheDocument();
	});

	it('SHOULD render the passed children', () => {
		render(<Text as='p'>Test content</Text>);

		expect(screen.getByText('Test content')).toBeInTheDocument();
	});

	it('SHOULD apply correct class names', () => {
		render(<Text as='p' color='yellow' size='36px' weight='700' uppercase className='custom-class' />);

		const textElement = screen.getByTestId('text');

		expect(textElement).toHaveClass('ui-text');
		expect(textElement).toHaveClass('color-yellow');
		expect(textElement).toHaveClass('weight-700');
		expect(textElement).toHaveClass('size-36px');
		expect(textElement).toHaveClass('uppercase');
		expect(textElement).toHaveClass('custom-class');
	});

	it.each<{tag: TextTag}>([{tag: 'h1'}, {tag: 'p'}, {tag: 'span'}, {tag: 'h4'}])(
		'SHOULD render correct tag - $tag',
		({tag}) => {
			const {getByTestId} = render(<Text as={tag} />);

			expect(getByTestId('text').tagName.toLowerCase()).toBe(tag);
		},
	);
});
