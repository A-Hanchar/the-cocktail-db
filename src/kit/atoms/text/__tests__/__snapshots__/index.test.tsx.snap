// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> SHOULD match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="ui-text color-base size-16px"
    data-test-id="text"
  />
</DocumentFragment>
`;
